src/transformers/models/clip/modeling_flax_clip.py

```diff
@@ -1007,6 +1037,79 @@ class FlaxCLIPTextModel(FlaxCLIPTextPreTrainedModel):
 )
 
 
+class FlaxCLIPTextModuleWithProjection(nn.Module):
+    config: CLIPTextConfig
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        self.text_model = FlaxCLIPTextTransformer(self.config, dtype=self.dtype)
+        self.text_projection = nn.Dense(self.config.projection_dim, use_bias=False)
```

Comment:
Projection layer should respect the `dtype` of the module
```suggestion
        self.text_projection = nn.Dense(self.config.projection_dim, use_bias=False, dtype=self.dtype)
```